SingularityService/src/main/java/com/hubspot/singularity/data/RequestManager.java

```diff
@@ -668,6 +736,13 @@ public void activateLeaderCache() {
       return webCache.getRequest(requestId);
     }
 
+    if (requestsCache.isEnabled() && !skipApiCache) {
+      SingularityRequestWithState request = requestsCache.get(requestId);
+      if (request != null) {
+        return Optional.of(request);
```

Comment:
I didn't make this `Optional.ofNullable(...)` in case it was a false null, aka the request came in between the 5 second reload, but this means that we will go to ZK for true nulls (the request was removed).